main/fmt.c

```diff
@@ -73,7 +73,7 @@ static int printTagField (fmtSpec* fspec, MIO* fp, const tagEntryInfo * tag)
 				break;
 		}
 
-		if (findex == tag->usedParserFields)
+		if (!f)
```

Comment:
@masatake why on [line 80](https://github.com/universal-ctags/ctags/pull/1635/files#diff-be3bbc112b04c745703ab241b76456d4R80) does it access `tag->parserFields [findex].ftype` rather than `f->ftype`?  wouldn't it be the same, and not depend on the value of `findex` nor on the internals of `tagEntryInfo` when there is `getParserField()` already?

anyway @jannick0 beware to keep `findex` the same so long as it's still used outside the loop -- I was about to suggest something like 

```C
 		for (findex = 0; ! f && findex < tag->usedParserFields; findex++)
 		{
            tagField *fitem = getParserField(tag, findex);
 			if (isParserFieldCompatibleWithFtype (fitem, ftype))
 				f = fitem;
 		}
```
above, but that changes the value of `findex` outside the loop, which is **_not_** acceptable here (yet).

Reply:
@b4n, I find it is my mistake.

`tag->parserFields [findex].ftype` should be `f->type`.
I forgot replacing it in 4e9e52c2d6903ac4704c84358d40b56b28c55d10.

Reply:
@b4n and @jannick0, I would like to fix what @b4n spotted separately from the change proposed in this pull request.